src/Storage/AuthorizationCodeStorage.php

```diff
@@ -16,29 +16,50 @@ class AuthorizationCodeStorage implements AuthorizationCodeInterface {
 		'id_token'     => 'string', // The OpenID Connect id_token.
 	);
 
-	public function getAuthorizationCode( $code ) {
+	private function getCodeMeta( $code ) {
+		global $wpdb;
+
 		if ( empty( $code ) ) {
 			return null;
 		}
 
-		$users = get_users(
-			array(
-				'number'     => 1,
-				'meta_key'   => self::META_KEY_PREFIX . '_code', // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_key
-				'meta_value' => $code, // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_value
+		$meta_row = $wpdb->get_row(
+			$wpdb->prepare(
+				"SELECT user_id, meta_key FROM $wpdb->usermeta WHERE meta_key LIKE '%s';",
+				'%' . $wpdb->esc_like( $code ),
```

Comment:
Did you see that there exists [a `meta_compare` parameter](http://developer.wordpress.org/reference/classes/wp_meta_query/) or is there another reason why you switched to a `$wpdb` query?

Reply:
Also need the `client_id` now that data is namespaced by client id, hence I am fetching meta_key here matching with code (which only uses the index) and then `*_user_meta()` calls

Reply:
Please don't do a like that starts with a `%`, it requires a table scan.